### PR TITLE
Decaffinate tests

### PR DESCRIPTION
**Pull request by realityking**
This PR does the following changes

* Decaffinate the tests
* Require Node.js 6
* Add eslint for linting

I'll back out the eslint change into its own PR after a first review as the license issue for `eslint-config-airbnb-base` still exists.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,13 @@
 language: node_js
 node_js:
-  - 4
   - 6
   - 8
   - 'stable'
+
+script:
+  - npm run lint
+  - npm run test
```
Comment (honzajavorek) [inline .travis.yml]: nitpick: `npm test` is sufficient

**Comment by realityking** [comment]
I split off linting into #21. Means no dependency changes 🎉 

I decaffeinated the actual parser in #22. Feel free to merge that one first, I'll then rebase this one.

Since releasing the decaffeinated code will be semver major (because of the bump to Node.js 6), it'd be great if you could checkout #16 as well.

**Comment by realityking** [comment]
@michalholasek Rebased!

**Comment by honzajavorek** [comment]
LGTM! Thank you!

**Pull request merged**